Remove unused function

mitchell-as opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2616" title="DX-2616" target="_blank"><img alt="Bug" src="https://activestatef.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10303?size=medium" />DX-2616</a>  Buildscript conflict resolution appears to get trumped by API
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

Looks like this was inadvertantly fixed in https://github.com/ActiveState/cli/pull/3249. I've updated the PR to just remove the unused code discovered during PR review.

mitchell-as commented on internal/runners/pull/pull.go:
```diff
@@ -272,7 +272,18 @@ func (p *Pull) mergeBuildScript(remoteCommit, localCommit strfmt.UUID) error {
+		return "", errs.Wrap(err, "Could not update build script with merged changes")
+	}
+
+	if synced, err := buildscriptRunbits.Sync(p.project, nil, p.out, p.auth); err == nil && synced {
+		return localcommit.Get(p.project.Dir())
+	} else {
+		return "", errs.Wrap(err, "Could not commit merged build script")
+	}
```
This change here may cause this ticket to be blocked on https://activestatef.atlassian.net/browse/DX-2619, where we want to do away with `buildscript.Sync()`.

mitchell-as commented on version.txt:
```diff
@@ -1 +1 @@
-0.45.0-RC1
\ No newline at end of file
+0.45.0-RC1
```
Needed to satisfy the "Check Format" action...

Naatan reviewed (changes requested):
In addition to comments, as noted on Slack; please make the buildscript being dirty an error condition for `state pull`, do not try to resolve conflicts with an uncommitted buildscript.

mitchell-as commented:
After addressing PR review comments and merging remote changes, it looks like this whole thing was inadvertantly fixed in https://github.com/ActiveState/cli/pull/3249. I've updated the PR to just remove the unused code discovered during PR review.

mitchell-as commented:
Please re-approve. I had to merge in v45's buildplan refactor as there was a conflict in this file related to file imports. I've resolve them here.
